### PR TITLE
Add option to run the tests against dendron rather than synapse.

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: lib/SyTest/Synapse.pm [hunk 1]
```diff
@@ -187,32 +205,66 @@ sub start
...
+      @synapse_command, "--generate-config", "--report-stats=no"
+   );
+
+   my @command;
+
+   if( $self->{dendron} ) {
+      $db_type eq "pg" or die "Dendron can only run against postgres";
+
+      my @db_arg_pairs = ();
+      while( my ( $key, $value ) = each %db_args ) {
```
Comment (leonerd) [inline lib/SyTest/Synapse.pm]: Don't use `each()` on a hash. It's silly and needs to be taken out and shot.

Instead, especially in this case, you might find `List::Util::pairmap` useful. It's like a regular `map` only takes its input list as an even-sized list of pairs, assigning values to `$a` and `$b` for each iteration:

    use List::Util qw( pairmap );

    my @db_arg_pairs = pairmap { $a eq "database" ? "dbname=$b" : "$a=$b" } %db_args;

File: lib/SyTest/Synapse.pm [hunk 2]
```diff
@@ -187,32 +205,66 @@ sub start
...
+         }
+         push @db_arg_pairs, $key . '=' . $value;
+      }
+
+      @command = (
+         $self->{dendron},
+         "--synapse-python" => $self->{python},
+         "--synapse-config" => $config_path,
+         "--synapse-url" => "http://127.0.0.1:$self->{unsecure_port}",
+         "--synapse-postgres" => join(" ", @db_arg_pairs),
```
Comment (leonerd) [inline lib/SyTest/Synapse.pm]: Whitespace `join( " ", @db_arg_pairs )`

Comment (NegativeMjark) [inline lib/SyTest/Synapse.pm]: Done dbcaef9

File: lib/SyTest/Synapse.pm (hunk 1 above)
Comment (NegativeMjark) [inline lib/SyTest/Synapse.pm]: Done dbcaef9

**Comment by leonerd** [comment]
LGTM

**Pull request merged**